src/goto-symex/symex_complexity.h

```diff
@@ -0,0 +1,155 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: John Dumbell
+
+\*******************************************************************/
+
+/// \file Symbolic Execution
+
+#ifndef CPROVER_GOTO_SYMEX_SYMEX_COMPLEXITY_H
+#define CPROVER_GOTO_SYMEX_SYMEX_COMPLEXITY_H
+
+#include "complexity_violation.h"
+#include "goto_symex_state.h"
+#include "symex_heuristic_transformation.h"
+
+class loop_complexity_infot
+{
+public:
+  loop_complexity_infot(irep_idt _loop_id, framet::loop_infot *_loop_info)
+    : loop_id(_loop_id), loop_info(_loop_info)
+  {
+  }
+
+  irep_idt loop_id;
+  framet::loop_infot *loop_info;
```

Comment:
Could we avoid a raw pointer here? Or at least document why it is safe to store it here.

Reply:
This is one of the main reasons I wanted someone with more C++ experience than me pitch in. Tried a few nicer options but couldn't work because of various reasons, so I'd like to know their preferred way of dealing with things like this.

Also the self-referencing pointer loop_infot too.

Reply:
I'm not exactly sure about what you need here, but seeing how it is used, I think you could use a frame iteraror + a loop_id and that should be enough to retrieve the info you need. The fact that the type is an iterator would make it clear that it shouldn't be used outside the scope of the container which was given as input to the function returning this.